src/metatrain/cli/train.py

```diff
@@ -406,12 +408,27 @@ def train_model(
     logger.info(
         f"Exporting model to `{output_checked}` and extensions to `{extensions_path}`"
     )
+    # get device from the model
+    device = next(
+        itertools.chain(
+            mts_atomistic_model.parameters(),
+            mts_atomistic_model.buffers(),
+        )
+    ).device
+    # always save on CPU. TODO: remove after release of
+    # https://github.com/lab-cosmo/metatensor/pull/668
+    mts_atomistic_model = mts_atomistic_model.to("cpu")
     mts_atomistic_model.save(str(output_checked), collect_extensions=extensions_path)
 
     ###########################
     # EVALUATE FINAL MODEL ####
     ###########################
 
+    mts_atomistic_model = load_atomistic_model(
+        str(output_checked), extensions_directory=extensions_path
+    )
```

Comment:
Why do we have to reload the model? I think it should be fine to just change the dtype agian to the one we want.

Reply:
It's good practice (see the torch tutorials). In this case I'm mainly doing it because `torch.jit.script` is only called during `MetatensorAtomisticModel.save()`. I.e., with the current metatensor, saving and reloading makes evaluation faster

Reply:
Okay, can you add a comment above the line stating that this is faster and is common practice even though it still looks weird to me.

Reply:
Commented